R/model.R

```diff
@@ -862,24 +864,98 @@ select_naomi_data <- function(
+
+  survey_vls_tagged <- survey_mf(survey_ids = vls_survey_ids,
                                 indicator = "viral_suppression_plhiv",
                                 survey_hiv_indicators = survey_hiv_indicators,
                                 naomi_mf = naomi_mf,
                                 use_kish = use_kish_vls,
                                 deff = deff_vls,
                                 use_aggregate = use_survey_aggregate)
 
-  naomi_mf$anc_prev_t1_dat <- anc_testing_prev_mf(anc_prev_year_t1, anc_testing, naomi_mf)
-  naomi_mf$anc_artcov_t1_dat <- anc_testing_artcov_mf(anc_artcov_year_t1, anc_testing, naomi_mf)
 
-  naomi_mf$anc_clients_t2_dat <- anc_testing_clients_mf(anc_clients_year_t2, anc_testing, naomi_mf,
-                                                        anc_clients_year_t2_num_months)
-  naomi_mf$anc_prev_t2_dat <- anc_testing_prev_mf(anc_prev_year_t2, anc_testing, naomi_mf)
-  naomi_mf$anc_artcov_t2_dat <- anc_testing_artcov_mf(anc_artcov_year_t2, anc_testing, naomi_mf)
+  # Aggregate, tag and subset ANC inputs according to model option specifications
+  if(is.null(anc_testing)) {
+    anc_model_mf <- NULL
+    anc_tagged <- NULL
+  } else {
+
+    anc_aggreagted <- aggregate_anc(anc_testing, naomi_mf$areas)
+
+    ## Calculate model inputs for all data provided
+    anc_full_mf <- anc_aggreagted %>%
+      dplyr::group_by(area_id, age_group, year) %>%
+      dplyr::summarise_at(dplyr::vars(anc_known_pos, anc_tested_pos, anc_tested,
+                                      anc_already_art, anc_clients, anc_known_neg,
+                                      births_facility), sum) %>%
+      dplyr::ungroup() %>%
+      dplyr::mutate(
+        sex = "female",
+        anc_total_pos = anc_known_pos + anc_tested_pos,
+        anc_prev_x = anc_known_pos + anc_tested_pos,
+        anc_prev_n = anc_known_pos + anc_tested + anc_known_neg,
+        anc_artcov_x = anc_already_art,
+        anc_artcov_n = anc_total_pos,
+        anc_clients_x = anc_clients,
+        anc_clients_pys_offset = log(anc_clients_year_t2_num_months/ 12),
+        anc_prevalence = anc_prev_x/anc_prev_n,
+        anc_art_coverage = anc_artcov_x/ anc_artcov_n) %>%
+      dplyr::select(area_id, age_group, year, sex, dplyr::everything()) %>%
+      tidyr::pivot_longer(cols = anc_known_pos:anc_art_coverage,
+                          names_to = "indicator", values_to = "value")
+
+    anc_years <- c(anc_prev_year_t1, anc_prev_year_t2,
+                   anc_artcov_year_t1, anc_artcov_year_t2,
+                   anc_clients_year_t2)
+
+    # Drop  NAs and filter for years contained in model options scope
+    anc_model_mf <- anc_full_mf %>%
+      dplyr::filter(
+        !is.na(value),
+        year %in% anc_years
+      ) %>%
+      dplyr::semi_join(
+        naomi_mf$mf_model,
+        by = "area_id"
+      )
+
+    anc_tagged <- tag_data_inputs(anc_full_mf, anc_model_mf, colnames(anc_model_mf))
```

Comment:
I am seeing this data frame with rows for indicator = `anc_clients_pys_offset`. Is that needed / supposed to be in here?

(See three rows from bottom in below excerpt)

``` 
> naomi_data$model_inputs$anc_full_mf %>% print(n = Inf)
# A tibble: 1,152 × 7
     area_id      age_group  year sex    indicator                    value naomi_input
     <chr>        <chr>     <int> <chr>  <chr>                        <dbl> <lgl>      
   1 MWI_2_1_demo Y015_049   2016 female anc_known_pos            1877      TRUE       
   2 MWI_2_1_demo Y015_049   2016 female anc_tested_pos           1741      TRUE       
   3 MWI_2_1_demo Y015_049   2016 female anc_tested              66672      TRUE       
   4 MWI_2_1_demo Y015_049   2016 female anc_already_art          1790      TRUE       
   5 MWI_2_1_demo Y015_049   2016 female anc_clients             75557      TRUE       
   6 MWI_2_1_demo Y015_049   2016 female anc_known_neg            1672      TRUE       
   7 MWI_2_1_demo Y015_049   2016 female births_facility         59852      TRUE       
   8 MWI_2_1_demo Y015_049   2016 female anc_total_pos            3618      TRUE       
   9 MWI_2_1_demo Y015_049   2016 female anc_prev_x               3618      TRUE       
  10 MWI_2_1_demo Y015_049   2016 female anc_prev_n              70221      TRUE       
  11 MWI_2_1_demo Y015_049   2016 female anc_artcov_x             1790      TRUE       
  12 MWI_2_1_demo Y015_049   2016 female anc_artcov_n             3618      TRUE       
  13 MWI_2_1_demo Y015_049   2016 female anc_clients_x           75557      TRUE       
  14 MWI_2_1_demo Y015_049   2016 female anc_clients_pys_offset     -0.288  TRUE       
  15 MWI_2_1_demo Y015_049   2016 female anc_prevalence              0.0515 TRUE       
  16 MWI_2_1_demo Y015_049   2016 female anc_art_coverage            0.495  TRUE       
```

Reply:
Needed to run model at aggregate level - not yet implemented